lib/floe/workflow/runner/docker.rb

```diff
@@ -73,8 +75,12 @@ def success?(runner_context)
         end
 
         def output(runner_context)
-          output = docker!("logs", runner_context["container_ref"], :combined_output => true).output
-          runner_context["output"] = output
+          runner_context["output"] ||= docker!("logs", runner_context["container_ref"], :combined_output => true).output
+        end
+
+        def mark_status(runner_context, type, error: true)
+          runner_context["container_state"] = {"Running" => false, "ExitCode" => error ? 999 : 0}
+          runner_context["output"] = type
```

Comment:
this is defining the internal values into the `runner_context`.

Instead, do we want to define `Error`, `Cause`, and `Output?
Or do we prefer to define these in the `Context#state` and skip the `runner`/`runner_context` all together?